src/reactThisBindingIssueRule.ts

```diff
@@ -53,6 +71,14 @@ class ReactThisBindingIssueRuleWalker extends ErrorTolerantWalker {
         this.getOptions().forEach((opt: any) => {
             if (typeof(opt) === 'object') {
                 this.allowAnonymousListeners = opt['allow-anonymous-listeners'] === true;
+                if (opt['bind-decorators']) {
+                    const allowedDecorators: any[] = opt['bind-decorators'];
+                    if (allowedDecorators.constructor !== Array && allowedDecorators.some((decorator) => typeof decorator !== 'string')) {
```

Comment:
`Array.isArray` should be more readable/convenien.
`||` makes more sense here. Because second check should be performed only if `opt['bind-decorators']` is an array (otherwise it will be runtime error)